src/monodroid/jni/android-system.cc

```diff
@@ -79,7 +89,51 @@ AndroidSystem::lookup_system_property (const char *name)
 			return p;
 	return nullptr;
 }
+#endif // DEBUG || !ANDROID
+
+const char*
+AndroidSystem::lookup_system_property (const char *name, uint32_t &value_len)
+{
+	value_len = 0;
+#if defined (DEBUG) || !defined (ANDROID)
+	BundledProperty *p = lookup_system_property (name);
+	if (p != nullptr) {
+		value_len = p->value_len;
+		return p->name;
+	}
+#endif // DEBUG || !ANDROID
+
+	if (application_config.system_property_count == 0)
+		return nullptr;
+
+	if (application_config.system_property_count % 2 != 0) {
+		log_warn (LOG_DEFAULT, "Corrupted environment variable array: does not contain an even number of entries (%u)", application_config.environment_variable_count);
+		return nullptr;
+	}
+
+	const char *prop_name;
+	const char *prop_value;
+	for (size_t i = 0; i < application_config.system_property_count; i += 2) {
```

Comment:
(high horse) and *here* is where negative numeric types are handy!  *IF* (lol?) there were `uint.MaxValue/2` environment variables, this would become an infinite loop, as we'd loop from `uint.MaxValue-2` to 0 without nary a blip.

(Granted, that would require having a lib with `uint.MaxValue/2` environment variables, which is *insane*...but kinda/sorts shows the utility of negative numbers. :-)

Reply:
They are useful **sometimes** :) But not in situations when you're supposed to specify a length of something, a size of something etc - that's where they are *insane* :)